extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java

```diff
@@ -80,15 +91,31 @@ void discoverBindableServices(BuildProducer<BindableServiceBuildItem> bindables,
                 }
             }
             bindables.produce(item);
+            grpcServiceNames.add(service.name());
         }
+
+        annotationTransformers.produce(new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
+            @Override
+            public boolean appliesTo(AnnotationTarget.Kind kind) {
+                return kind == AnnotationTarget.Kind.METHOD;
+            }
+
+            @Override
+            public void transform(TransformationContext transformationContext) {
+                MethodInfo target = transformationContext.getTarget().asMethod();
+                if (grpcServiceNames.contains(target.declaringClass().name()) && !"bindService".equals(target.name())) {
+                    transformationContext.transform().add(CleanUpRequestContext.class).done();
```

Comment:
We could add the interceptor binding directly to the `@GrpcService` stereotype and only transform the classes that are annotated with `@Singleton`.

Reply:
Hm, is there a reason why we don't add the binding on the class level? I've noticed some warnings like these:
```
2021-04-30 13:49:20,719 WARN  [io.qua.arc.dep.sta.InterceptedStaticMethodsProcessor] (build-11) Interception of private static methods is not supported; bindings found on io.quarkus.grpc.server.services.MutinyHelloService: io.grpc.examples.helloworld.HelloReply lambda$sayHello$1(java.lang.String s)
```
If we need to transform the methods then we should probably skip at least private, static and synthetic methods.

Reply:
@cescoffier what do you think of requiring users to use `@GrpcService`? That would be even more breaking but since we are already breaking...

Then we wouldn't have to do any annotation transformations here